### DISCUSSION TITLE
"Binaries built for a non-native architecture"

### DISCUSSION THREAD

**Discussion by btb** [Homebrew/discussions#2005] [Writing Formulae/Casks] [answered]
I'm trying to make a tap and bottles for https://github.com/open-watcom/open-watcom-v2, following the guide here: https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

But brew test-bot fails with "Error: Binaries built for a non-native architecture were installed into open-watcom-v2's prefix."
Which, yes, that's the point, this is for doing cross-compiling and cross-platform testing. Is there a way around this?

**Comment by ipatch** [comment]
i'm far from being a tap maintainer expert, but linking to the homebrew tap that you've created would help others understand what is going on.

you link to a project repo that isn't a homebrew tap, so if you want to bundle that project, you'll have to create a homebrew tap as in the article you linked discusses. however the github environment only provides catalina as a build environment.

so you want to build for m1 which i'm presuming you do, then you'll have to create a self-hosted runner to build your formulae on. because as of right now (aug 18, 2021) github or any service i know of do not provider runner/envs for building m1 binaries.

**Comment by btb** [comment]
My tap is in its own repo here: https://github.com/btb/homebrew-open-watcom

Not for m1 (yet), but for running on x64 linux or mac to cross-develop for dos and windows.

I have it working to build the bare binaries for the native platform, but when I try to build the whole package needed (here: https://github.com/btb/homebrew-open-watcom/pull/7) it fails as above.

**Comment by carlocab** [comment, marked as answer]
Add the formula to `audit_exceptions/mismatched_binary_allowlist.json`, like so: https://github.com/Homebrew/homebrew-core/blob/master/audit_exceptions/mismatched_binary_allowlist.json

**Comment by btb** [comment]
Ah, perfect, I thought there must be something like that. 

But I added it to the PR's branch and it's failing with:
```
==> FAILED
14btb/open-watcom:
15  * audit_exceptions/mismatched_binary_allowlist.json references
16    formulae or casks that are not found in the btb/open-watcom tap.
17    Invalid formulae or casks: open-watcom-v2
```

Still trying things, but did I mess up my tap somehow to cause this?

**Reply by alebcay** [reply]
In the allowlist, try adding `btb/open-watcom/open-watcom-v2` (the fully qualified name of the formula).

**Reply by Rylan12** [reply]
This error is due to a bug that is fixed by https://github.com/Homebrew/brew/pull/11895. Once that's merged, you'll need to go back to using the short name. The full names aren't able to be used in the exception lists

**Reply by btb** [reply]
it works perfectly now. Thanks @carlocab  and @Rylan12 !
